github/transport.go

```diff
@@ -59,7 +60,7 @@ func (rlt *RateLimitTransport) RoundTrip(req *http.Request) (*http.Response, err
 	// Make requests for a single user or client ID serially
 	// This is also necessary for safely saving
 	// and restoring bodies between retries below
-	rlt.lock(req)
+	rlt.smartLock(true)
```

Comment:
Can you describe what this change is doing?

Reply:
Updated the comment before the first smartLock occurrence and the function.